File: DocumentAnalysisSystem/Visitors/ParagraphCountVisitor.cs
using DocumentAnalysisSystem.Models;

namespace DocumentAnalysisSystem.Visitors
{
    public class ParagraphCountVisitor : IDocumentVisitor
    {
        public int ParagraphCount { get; private set; }

        public void Visit(Document document)
        {
            // No count at document level.
        }

        public void Visit(Section section)
        {
            // No count at section level.
        }

        public void Visit(Paragraph paragraph)
        {
            ParagraphCount++;
        }
    }
}

File: DocumentAnalysisSystemJava/src/main/java/com/documentanalysis/visitors/ParagraphCountVisitor.java
package com.documentanalysis.visitors;

import com.documentanalysis.models.Document;
import com.documentanalysis.models.Paragraph;
import com.documentanalysis.models.Section;

public class ParagraphCountVisitor implements IDocumentVisitor {
    private int paragraphCount;

    public int getParagraphCount() {
        return paragraphCount;
    }

    @Override
    public void visit(Document document) {
        // No direct count at document level.
    }

    @Override
    public void visit(Section section) {
        // No direct count at section level.
    }

    @Override
    public void visit(Paragraph paragraph) {
        paragraphCount++;
    }
}
